library/azure_rm_trafficmanagerendpoint.py

```diff
@@ -0,0 +1,358 @@
+        type: str
+    min_child_endpoints:
+        description:
+            - The minimum number of endpoints that must be available in the child profile in order for the parent profile to be considered available.
+            - Only applicable to endpoint of type 'NestedEndpoints'.
+        type: int
+    geo_mapping:
+        description:
+            - The list of countries/regions mapped to this endpoint when using the 'Geographic' traffic routing method.
+        type: str
+    state:
+        description:
+            - Assert the state of the Traffic Manager endpoint. Use C(present) to create or update a Traffic Manager endpoint and C(absent) to delete it.
+        default: present
+        choices:
+            - absent
+            - present
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Hai Cao <t-haicao@microsoft.com>"
+    - "Yunge Zhu <yungez@microsoft.com>"
+
+'''
+
+EXAMPLES = '''
+
+'''
+RETURN = '''
+'''
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+    from azure.mgmt.trafficmanager.models import (
+        Endpoint, DnsConfig, MonitorConfig
+    )
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+def traffic_manager_endpoint_to_dict(endpoint):
+    return dict(
+        id=endpoint.id,
+        name=endpoint.name,
+        type=endpoint.type,
+        target_resource_id=endpoint.target_resource_id,
+        target=endpoint.target,
+        status=endpoint.endpoint_status,
+        weight=endpoint.weight,
+        priority=endpoint.priority,
+        location=endpoint.endpoint_location,
+        monitor_status=endpoint.endpoint_monitor_status,
+        min_child_endpoints=endpoint.min_child_endpoints,
+        geo_mapping=endpoint.geo_mapping
+    )
+    # if endpoint.geo_mapping:
+
+    #     for geo in endpoint.geo_mapping:
+    #         result['geo_mapping'].append(geo)
+    )
+
+
+class Actions:
+    NoAction, CreateOrUpdate, Delete = range(3)
+
+class AzureRMTrafficManagerEndpoint(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            profile_name=dict(
+                type='str',
+                required=True
+            ),
+            type=dict(
+                type='str',
+                choices=['azureEndpoints', 'externalEndpoints', 'nestedEndpoints'],
+                required=True
+            ),
+            target_resource_id=dict(type='str'),
+            enable=dict(type='bool', default=True),
+            weight=dict(type='int'),
+            priority=dict(type='int'),
+            location=dict(type='str'),
+            minitor_status=dict(type='str'),
```

Comment:
Endpoint monitor status cannot be changed manually. It's a combination of the results of endpoint monitoring and the configured endpoint status. link for ref: https://docs.microsoft.com/en-us/azure/traffic-manager/traffic-manager-monitoring#endpoint-monitor-status
It also should not be check in check_update()